sympy/core/basic.py

```diff
@@ -2091,7 +2091,7 @@ def class_key(cls):
 
     @cacheit
     def sort_key(self, order=None):
-        return self.class_key(), (1, (str(self),)), S.One.sort_key(), S.One
+        return self.class_key(), (1, (str(self),)), S.One.sort_key(), S.One.sort_key()
```

Comment:
Maybe if we have `class_key` we actually don't need to call `sort_key` in these methods at all.

Reply:
Maybe something else is needed to handle the coefficient problem referred to above like a special case to handle when the coefficient is just `1`.